doc/api/node_report.md

```diff
@@ -0,0 +1,116 @@
+# node-report
+
+Delivers a human-readable diagnostic summary, written to file
+or retrieved as a text.
+
+The report is intended for development, test and production
+use, to capture and preserve information for problem determination.
+It includes JavaScript and native stack traces, heap statistics,
+platform information and resource usage etc. With the report enabled,
+reports can be triggered on unhandled exceptions, fatal errors, signals.
+
+Many capabilities are available as APIs too, for being consumed in-process.
+
+Iin-built node-report function is supported in Node.js versions 11.0.0 onwards.
+For Node.js versions 8 and below, please use it's [npm counterpart][] instead.
+
+## Usage
+
+```bash
+node --report-events fatalerror,signal,exception app.js
+```
+A report will be triggered automatically on unhandled exceptions and fatal
+error events (for example out of memory errors), and can also be triggered
+by sending a USR2 signal to a Node.js process (not supported on Windows).
+
+A report can also be triggered via an API call from a JavaScript
+application.
+
+```js
+const util = require('util');
+util.triggerReport();
+```
+The content of a report can also be returned as a JavaScript string via an
+API call from a JavaScript application.
+
+```js
+const util = require('util');
+const report_str = util.getReport();
+console.log(report_str);
+```
+
+These APIs can be used without adding the automatic exception
+and fatal error hooks and the signal handler.
+
+Content of the report consists of a header section containing the event
+type, date, time, PID and Node version, sections containing JavaScript and
+native stack traces, a section containing V8 heap information, a section
+containing libuv handle information and an OS platform information section
+showing CPU and memory usage and system limits. An example report can be
+triggered using the Node.js REPL:
+
+```raw
+$ node
+> const util = require('util');
+> util.triggerReport();
+Writing Node.js report to file: node-report.20180820.091102.8480.001.txt
+Node.js report completed
+>
+```
+
+When a report is triggered, start and end messages are issued to stderr
+and the filename of the report is returned to the caller. The default filename
+includes the date, time, PID and a sequence number. Alternatively, a filename
+can be specified as a parameter on the `triggerReport()` call.
```

Comment:
Does this filename conform to typical `require('fs')` path rules? For instance, could I pass in a file URL here? `triggerReport(new URL('file:///whatever')`. If there are differences between this and what `fs` accepts, or if there are no differences, it should be noted here.

Reply:
The file name will be treated as file name in the local fs and no translation occurs on the passed string. Documentation is updated. quoting the same.

```+ filename specifies the name of the output file in the file system```
```+ No translations occur to the supplied string, so URL protocols```
```+ are not supported.```